File: frontend/src/api/docs.ts
/**
 * 文档浏览 API
 */

import api from './index'

/**
 * 文档文件
 */
export interface DocFile {
  name: string
  path: string
  title: string
}

/**
 * 文档分类
 */
export interface DocCategory {
  id: string
  name: string
  icon: string
  description: string
  files: DocFile[]
}

/**
 * 文档结构
 */
export interface DocsStructure {
  categories: DocCategory[]
  total_files: number
}

/**
 * 文档内容
 */
export interface DocContent {
  path: string
  title: string
  content: string
  category: string
}

/**
 * 获取文档目录结构
 */
export async function getDocsStructure(): Promise<DocsStructure> {
  const response = await api.get<{ success: boolean; data: DocsStructure }>('/v1/docs/structure')
  return response.data.data
}

/**
 * 获取文档内容
 * @param docPath - 文档路径
 */
export async function getDocContent(docPath: string): Promise<DocContent> {
  const response = await api.get<{ success: boolean; data: DocContent }>(`/v1/docs/content/${docPath}`)
  return response.data.data
}

File: models/docs.py
"""
文档浏览相关数据模型
"""

from typing import List, Optional
from pydantic import BaseModel, Field


class DocFile(BaseModel):
    """文档文件"""
    name: str = Field(..., description="文件名")
    path: str = Field(..., description="相对路径")
    title: str = Field(..., description="文档标题")


class DocCategory(BaseModel):
    """文档分类"""
    id: str = Field(..., description="分类 ID")
    name: str = Field(..., description="分类名称")
    icon: str = Field(default="📄", description="图标")
    description: str = Field(default="", description="分类描述")
    files: List[DocFile] = Field(default_factory=list, description="文件列表")


class DocsStructure(BaseModel):
    """文档结构"""
    categories: List[DocCategory] = Field(..., description="分类列表")
    total_files: int = Field(..., description="总文件数")


class DocContent(BaseModel):
    """文档内容"""
    path: str = Field(..., description="文档路径")
    title: str = Field(..., description="文档标题")
    content: str = Field(..., description="Markdown 内容")
    category: str = Field(..., description="所属分类")


class DocsStructureResponse(BaseModel):
    """文档结构响应"""
    success: bool = True
    data: DocsStructure


class DocContentResponse(BaseModel):
    """文档内容响应"""
    success: bool = True
    data: DocContent
